ros2pkg/ros2pkg/api/create_api.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2017 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import em
+from io import StringIO
+import os
+import sys
+
+
+def expand_template(template_file, data, output_file, minimum_timestamp=None):
+    output = StringIO()
+    interpreter = em.Interpreter(
+        output=output,
+        options={
+            em.BUFFERED_OPT: True,
+            em.RAW_OPT: True,
+        },
+        globals=data,
+    )
+    with open(template_file, 'r') as h:
+        try:
+            interpreter.file(h)
+        except Exception:
+            if os.path.exists(output_file):
+                os.remove(output_file)
+            print("Exception when expanding '%s' into '%s'" %
+                  (template_file, output_file), file=sys.stderr)
+            raise
+    content = output.getvalue()
+    interpreter.shutdown()
```

Comment:
Since it is only used in one location that is quite some block of code (without three lines `try`, `finally`, and `shutdown` would be enough). But it is a "correct" way in Python. Anyone reading the code will have to look up what this custom class does rather than directly understand the code where it is being used. So I would slightly lean to avoid a custom class it in this case.

Beside that I would argue though that you shouldn't embed the constructor arguments into the custom class since it prevents reuse. Also the constructor signature of your custom class is different from the base class which is confusing to the user imo.